### PR TITLE
chore(composer): Move executor task and (gRPC and Geth)collectors to the Composer

### PR DESCRIPTION
**Pull request by bharath-123**
## Summary
This is the final PR of a wholesale refactor of Composer where we move the Executor and (gRPC and Geth)Collector tasks to the Composer top level.

## Background
Previously, Collectors and Executors were part of a higher level Searcher task i.e they were leaf'ed of the Composer task. As we were adding the gRPC collector, this made the testing of collectors much harder as they were stemming away from the Composer.

## Changes
- Spawn the Executor and Collector tasks off Composer. 
- Move Collectors, Executor related data fields to the Composer
- Remove unnecessary code from Searcher struct and Executor struct.
 
## Testing
Since this refactor is adding no functional code but is just a refactor. We can validate it by making sure the codebase compiles and all the tests successfully runs.

## Issues
#795

### CODE REVIEW

File: crates/astria-composer/src/composer.rs [hunk 1]
```diff
@@ -37,20 +78,70 @@ impl Composer {
...
-    pub fn from_config(cfg: &Config) -> eyre::Result<Self> {
-        let searcher = Searcher::from_config(cfg).wrap_err("failed to initialize searcher")?;
+    pub async fn from_config(cfg: &Config) -> eyre::Result<Self> {
+        let (serialized_rollup_transactions_tx, serialized_rollup_transactions_rx) =
+            tokio::sync::mpsc::channel(256);
+        let (searcher_status_sender, _) = watch::channel(Status::default());
+
+        let executor_handle = ExecutorHandle {
+            sequence_action_tx: serialized_rollup_transactions_tx.clone(),
+        };
```
Comment (bharath-123) [inline crates/astria-composer/src/composer.rs:88]: Let executor create the handle and creates it for you.

File: crates/astria-composer/src/searcher/executor/mod.rs [hunk 2]
```diff
@@ -110,8 +110,8 @@ impl Drop for Executor {
 }
 
 #[derive(Debug)]
-pub(super) struct Status {
-    is_connected: bool,
+pub(crate) struct Status {
+    pub(crate) is_connected: bool,
```
Comment (bharath-123) [inline crates/astria-composer/src/searcher/executor/mod.rs:114]: use `pub(super)` and expose data with getters and setters

**Comment by bharath-123** [comment]
Flatten the file structure of searcher and executor.

File: crates/astria-composer/Cargo.toml [hunk 3]
```diff
@@ -13,7 +13,7 @@ name = "astria-composer"
 
 [dependencies]
 astria-build-info = { path = "../astria-build-info", features = ["runtime"] }
-astria-core = { path = "../astria-core" }
+astria-core = { path = "../astria-core", features = ["client", "server"] }
```
Comment (noot) [inline crates/astria-composer/Cargo.toml:16]: put the `client` feature under `dev-dependencies` only

File: crates/astria-composer/src/composer.rs [hunk 4]
```diff
@@ -15,19 +31,44 @@ use crate::{
...
 pub struct Composer {
-    /// ApiServer is used for monitoring status of the Composer service.
+    /// `ApiServer` is used for monitoring status of the Composer service.
     api_server: ApiServer,
-    /// Searcher establishes connections to individual rollup nodes, receiving
+    /// `Searcher` establishes connections to individual rollup nodes, receiving
     /// pending transactions from them and wraps them as sequencer transactions
     /// for submission.
-    searcher: Searcher,
+    // searcher: Searcher,
```
Comment (noot) [inline crates/astria-composer/src/composer.rs:53]: remove

File: crates/astria-composer/src/composer.rs [hunk 5]
```diff
@@ -59,26 +150,177 @@ impl Composer {
...
+    /// Returns an error if the listener is not bound
+    pub fn grpc_collector_local_addr(&self) -> io::Result<SocketAddr> {
+        self.grpc_collector_listener.local_addr()
+    }
+
     /// Runs the composer.
     ///
     /// Currently only exits if the api server or searcher stop unexpectedly.
-    pub async fn run_until_stopped(self) {
+    /// # Errors
```
Comment (noot) [inline crates/astria-composer/src/composer.rs:163]: mention the errors?

File: crates/astria-composer/src/composer_service.rs [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+        if submit_sequence_actions_request.sequence_actions.is_empty() {
+            return Err(tonic::Status::invalid_argument(
+                "No sequence actions provided",
+            ));
+        }
+
+        // package the sequence actions into a SequenceAction and send it to the searcher
+        for sequence_action in submit_sequence_actions_request.sequence_actions {
+            let sequence_action = SequenceAction {
+                rollup_id: RollupId::from_unhashed_bytes(sequence_action.rollup_id),
```
Comment (noot) [inline crates/astria-composer/src/composer_service.rs:44]: the rollup ID is already hashed, so should use `RollupId::try_from_slice`

File: crates/astria-composer/src/searcher/geth_collector.rs [hunk 7]
```diff
@@ -84,15 +84,15 @@ impl Collector {
         }
     }
 
-    /// Subscribe to the collector's status.
-    pub(super) fn subscribe(&self) -> watch::Receiver<Status> {
+    /// Subscribe to the composer's status.
```
Comment (noot) [inline crates/astria-composer/src/searcher/geth_collector.rs:87]: revert comment change

**Comment by bharath-123** [comment]
Closing this as per a better PR re-structure as suggested by Janis.

**Pull request closed without merging**